core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerOutputPathResolver.java

```diff
@@ -26,23 +26,39 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.pentaho.platform.api.engine.IPentahoSession;
+import org.pentaho.platform.api.genericfile.GenericFilePermission;
 import org.pentaho.platform.api.genericfile.IGenericFileService;
+import org.pentaho.platform.api.genericfile.exception.OperationFailedException;
 import org.pentaho.platform.api.usersettings.IUserSettingService;
 import org.pentaho.platform.api.usersettings.pojo.IUserSetting;
 import org.pentaho.platform.engine.core.system.PentahoSessionHolder;
 import org.pentaho.platform.engine.core.system.PentahoSystem;
 import org.pentaho.platform.repository.RepositoryFilenameUtils;
 import org.pentaho.platform.repository2.ClientRepositoryPaths;
+import org.pentaho.platform.scheduler2.messsages.Messages;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
 
 /**
  * @author Rowell Belen
  */
 public class SchedulerOutputPathResolver {
 
-  final String DEFAULT_SETTING_KEY = "default-scheduler-output-path";
+  private static final String DEFAULT_SETTING_KEY = "default-scheduler-output-path";
 
   private static final Log logger = LogFactory.getLog( SchedulerOutputPathResolver.class );
 
+  private static final List<GenericFilePermission> permissions = new ArrayList<>();
+
+  static {
+    // initialize permissions
+    permissions.add( GenericFilePermission.READ );
+    permissions.add( GenericFilePermission.WRITE );
+  }
```

Comment:
This class was not validating access to the output path, causing invalid output paths to not be detected when creating or updating a schedule. This could happen with PVFS connections, given the Select Folder dialog currently does not check the validity of PVFS connection root folders, when selected. The invalid output path would only be detected when executing the schedule, when passing through `action/SchedulerOutputPathResolver`.